waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -508,28 +513,33 @@ proc getNumStreams*(pm: PeerManager, protocol: string): (int, int) =
               numStreamsOut += 1
   return (numStreamsIn, numStreamsOut)
 
+proc pruneInRelayConns(pm: PeerManager, amount: int) {.async.} =
+  let (inRelayPeers, outRelayPeers) = pm.connectedPeers(WakuRelayCodec)
+  let connsToPrune = min(amount, inRelayPeers.len)
+
+  for p in inRelayPeers[0..<connsToPrune]:
+    await pm.switch.disconnect(p)
+
 proc connectToRelayPeers*(pm: PeerManager) {.async.} =
+  let (inRelayPeers, outRelayPeers) = pm.connectedPeers(WakuRelayCodec)
   let maxConnections = pm.switch.connManager.inSema.size
-  let (inRelayPeers, outRelayPeers) = pm.getNumConnections(WakuRelayCodec)
-  let totalRelayPeers = inRelayPeers + outRelayPeers
+  let totalRelayPeers = inRelayPeers.len + outRelayPeers.len
+  let inPeersTarget = maxConnections - pm.outPeersTarget
+
+  if inRelayPeers.len > inPeersTarget:
+    await pm.pruneInRelayConns(inRelayPeers.len-inPeersTarget)
+
+  if outRelayPeers.len >= pm.outPeersTarget:
+    return
```

Comment:
Shall we also prune the exceeding `out` connection at this point?

Reply:
good point, but the thing is that since we control the out connections that we create, that shouldnt be necessary. i mean, we should never exceed the max out limit, since we do this check at `connectToRelayPeers`.

note that on top of this in prunning, we will have a prunning based on libp2p scoring, which will prune in/out connections from our mesh based on a given score. But this comes from gossipsub.